src/os/bluestore/NVMEDevice.cc

```diff
@@ -1007,14 +1003,11 @@ int NVMEDevice::read_random(uint64_t off, uint64_t len, char *buf, bool buffered
   ++ioc.num_reading;
   driver->queue_task(t);
 
-  {
-    std::unique_lock<std::mutex> l(ioc.lock);
-    while (t->return_code > 0)
-      ioc.cond.wait(l);
+  while (t->return_code > 0) {
+	ioc.aio_wait();
```

Comment:
this isn't a safe check. we may enter the loop while return_code == 0, but before wait other thread change this and signal. at last we are stuck into wait infinitely!